### PR TITLE
Add Scoop

### PR DESCRIPTION
**Pull request by NatoBoram**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

---

* Requires https://github.com/microsoft/winget-cli/issues/298
* Requires https://github.com/microsoft/winget-cli/issues/182
* Requires https://github.com/microsoft/winget-pkgs/issues/101699

```yaml
InstallerType: ps1
```

Scoop is a command-line installer for Windows.

* Old PR : https://github.com/microsoft/winget-pkgs/pull/825

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/19425)

### CODE REVIEW

**Comment by NatoBoram** [comment]
Hello bot! It's okay, https://github.com/microsoft/winget-cli/issues/298 isn't resolved yet.

**Comment by denelon** [comment]
I added "Blocking Issue" so the bot doesn't try and close this one in 14 days :)

**Review by ItzLevvie — DISMISSED** [review]

**Comment by denelon** [comment]
@NatoBoram we've added support for portable packages. It might be worth testing with a local 1.2 manifest to ensure this works. You can use either [wingetcreate](https://github.com/microsoft/winget-create) or the [YamlCreate](https://github.com/microsoft/winget-pkgs/blob/master/Tools/YamlCreate.ps1) script to build the manifest.

**Comment by NatoBoram** [comment]
I managed to use YamlCreate to get it to the manifest version 1.2.0, but the "publisher name" changed so I guess I'll have to re-run the script to change that.

In the meantime, I installed it using the `InstallerType: portable` but it creates a file at `C:\Users\natob\AppData\Local\Microsoft\WinGet\Links\scoop.exe`. It's probably great for portable apps but this is a [PowerShell installer](https://github.com/ScoopInstaller/Install/blob/master/install.ps1) for a [full-blown PowerShell project](https://github.com/ScoopInstaller/Scoop/tree/master/lib)

**Review by ItzLevvie — APPROVED** [review]

**Comment by denelon** [comment]
> I managed to use YamlCreate to get it to the manifest version 1.2.0, but the "publisher name" changed so I guess I'll have to re-run the script to change that.
> 
> In the meantime, I installed it using the `InstallerType: portable` but it creates a file at `C:\Users\natob\AppData\Local\Microsoft\WinGet\Links\scoop.exe`. It's probably great for portable apps but this is a [PowerShell installer](https://github.com/ScoopInstaller/Install/blob/master/install.ps1) for a [full-blown PowerShell project](https://github.com/ScoopInstaller/Scoop/tree/master/lib). I guess a new installer type is still required?

I see. 

We're still working through the implications of PowerShell based solutions. One of or design tenets has been to avoid scripting. It can be quite tricky to test/validate from a security and compliance standpoint. I'm guessing there isn't a standalone executable or another form of installer for scoop then.

**Review by ItzLevvie — DISMISSED** [review]

**Comment by stephengillie** [comment]
Manual Validation failed with:
> 2023-04-04 13:42:05.221 [CLI ] Package hash verification failed. SHA256 in manifest [0f2dc9aa7e8d01ee5af1d1a31483d64ad3ecd85ee4232c08686c44dbc832ac4c] does not match download [5ac8b2876e83b758dfd44b87033ac0e2a049fd43b4f059a6739008aa547bf0f0]

File: manifests/l/LukeSampson/Scoop/0.2.4/LukeSampson.Scoop.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+InstallerSwitches:
+  Silent: " | Out-Null"
+  SilentWithProgress:
+UpgradeBehavior: install
+Commands:
+  - scoop
+Installers:
+  - Architecture: neutral
+    InstallerUrl: https://get.scoop.sh
+    InstallerSha256: 0F2DC9AA7E8D01EE5AF1D1A31483D64AD3ECD85EE4232C08686C44DBC832AC4C
```
Comment (stephengillie) [inline manifests/l/LukeSampson/Scoop/0.2.4/LukeSampson.Scoop.installer.yaml:20]:
```suggestion
    InstallerSha256: 5AC8B2876E83B758DFD44B87033AC0E2A049FD43B4F059A6739008AA547BF0F0
```

File: manifests/l/LukeSampson/Scoop/0.2.4/LukeSampson.Scoop.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Created with YamlCreate.ps1 v2.1.3 $debug=MDSU.7-3-0-preview-6
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json
+
+PackageIdentifier: LukeSampson.Scoop
+PackageVersion: 0.2.4
```
Comment (NatoBoram) [inline manifests/l/LukeSampson/Scoop/0.2.4/LukeSampson.Scoop.installer.yaml:5]:
```suggestion
PackageVersion: 0.3.1
```

File: manifests/l/LukeSampson/Scoop/0.2.4/LukeSampson.Scoop.locale.en-US.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# Created with YamlCreate.ps1 v2.1.3 $debug=MDSU.7-3-0-preview-6
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: LukeSampson.Scoop
+PackageVersion: 0.2.4
```
Comment (NatoBoram) [inline manifests/l/LukeSampson/Scoop/0.2.4/LukeSampson.Scoop.locale.en-US.yaml:5]:
```suggestion
PackageVersion: 0.3.1
```

**Comment by stephengillie** [comment]
Automatic Validation failed with:
>  Manifest: The manifest file path must match PackageIdentifier and PackageVersion properties from manifest. actual 'manifests\l\LukeSampson\Scoop\0.2.4', expected 'manifests\l\LukeSampson\Scoop\0.3.1' 

Can the file paths be updated to match the version?

**Comment by NatoBoram** [comment]
Oh, sorry about that

**Comment by stephengillie** [comment]
Getting this error consistently, in different VMs and pipelines, for numerous executables:
> C:\Users\validator\AppData\Local\Microsoft\WinGet\Links\scoop.exe Exception during executable launch operation System.AggregateException: One or more errors occurred. (The specified executable is not a valid application for this OS platform.) 

Are these for `arm` or `arm64` instead of `x64`?

**Comment by NatoBoram** [comment]
It's actually `ps1`

> I installed it using the `InstallerType: portable` but it creates a file at `C:\Users\natob\AppData\Local\Microsoft\WinGet\Links\scoop.exe`. It's probably great for portable apps but this is a [PowerShell installer](https://github.com/ScoopInstaller/Install/blob/master/install.ps1) for a [full-blown PowerShell project](https://github.com/ScoopInstaller/Scoop/tree/master/lib). I guess a new installer type is still required?

**Comment by denelon** [comment]
@NatoBoram scripts aren't supported. We need to add additional logic to the validation infrastructure to check the extension of the downloaded file and provide an appropriate message for validation.

**Review by ItzLevvie — APPROVED** [review]

**Comment by NatoBoram** [comment]
Been a while since I haven't booted Windows, oh my.

@stephengillie @denelon scripts are still unsupported, I imagine? Do we need to re-set https://github.com/microsoft/winget-pkgs/labels/Blocking-Issue?

**Comment by stephengillie** [comment]
Hi @NatoBoram,

Apologies - I re-ran our oldest PRs, and accidentally included this PR.

**Comment by denelon** [comment]
The community repository is going to require an installer rather than a script. I'm going to go ahead and close this PR.

**Comment by NatoBoram** [comment]
I guess https://github.com/microsoft/winget-pkgs/issues/101699 will need to be closed then?

**Pull request closed without merging**